### ISSUE TITLE
Getting error SSLConnectionSocketFactory not found in packages?

### ISSUE THREAD

**Issue by shubhamjaincav** [elastic/logstash#12275]
Hi,

i am getting error in logstash 7.4 and showing error like :

NameError: SSLConnectionSocketFactory not found in packages org.apache.http.client.methods, org.apache.http.client.entity, org.apache.http.client.config, org.apache.http.config, org.apache.http.conn.socket, org.apache.http.impl, org.apache.http.impl.client, org.apache.http.impl.conn, org.apache.http.impl.auth, org.apache.http.entity, org.apache.http.message, org.apache.http.params, org.apache.http.protocol, org.apache.http.auth, java.util.concurrent, org.apache.http.client.protocol, org.apache.http.conn.ssl, java.security.cert, java.security.spec, java.security, org.apache.http.client.utils; last error: cannot load Java class org.apache.http.client.utils.SSLConnectionSocketFactory>, :backtrace=>["uri:classloader:/jruby/java/core_ext/module.rb:45:in `const_missing'"

vendor/bundle/jruby/2.5.0/gems/manticore-0.6.4-java/lib/manticore/client.rb:606:in `ssl_socket_factory_from_options'

full stack:

[2020-09-25T11:57:32,241][ERROR][logstash.javapipeline    ][main] Pipeline aborted due to error {:pipeline_id=>"main", :exception=>#<NameError: SSLConnectionSocketFactory not found in packages org.apache.http.client.methods, org.apache.http.client.entity, org.apache.http.client.cooptionig, org.apache.http.cooptionig, org.apache.http.conn.socket, org.apache.http.impl, org.apache.http.impl.client, org.apache.http.impl.conn, org.apache.http.impl.auth, org.apache.http.entity, org.apache.http.message, org.apache.http.params, org.apache.http.protocol, org.apache.http.auth, java.util.concurrent, org.apache.http.client.protocol, org.apache.http.conn.ssl, java.security.cert, java.security.spec, java.security, org.apache.http.client.utils; last error: cannot load Java class org.apache.http.client.utils.SSLConnectionSocketFactory>, :backtrace=>["uri:classloader:/jruby/java/core_ext/module.rb:45:in `const_missing'", "/opt/home//option/all/vendor/bundle/jruby/2.5.0/gems/manticore-0.6.4-java/lib/manticore/client.rb:606:in `ssl_socket_factory_from_options'", "/opt/home//option/all/vendor/bundle/jruby/2.5.0/gems/manticore-0.6.4-java/lib/manticore/client.rb:397:in `pool_builder'", "/opt/home//option/all/vendor/bundle/jruby/2.5.0/gems/manticore-0.6.4-java/lib/manticore/client.rb:405:in `pool'", "/opt/home//option/all/vendor/bundle/jruby/2.5.0/gems/manticore-0.6.4-java/lib/manticore/client.rb:209:in `initialize'", "/opt/home//option/all/vendor/gems/71a5254b/logstash-output-elasticsearch-10.1.0-java/lib/logstash/outputs/elasticsearch/http_client/manticore_adapter.rb:26:in `initialize'", "/opt/home//option/all/vendor/gems/71a5254b/logstash-output-elasticsearch-10.1.0-java/lib/logstash/outputs/elasticsearch/http_client.rb:330:in `build_adapter'", "/opt/home//option/all/vendor/gems/71a5254b/logstash-output-elasticsearch-10.1.0-java/lib/logstash/outputs/elasticsearch/http_client.rb:334:in `build_pool'", "/opt/home//option/all/vendor/gems/71a5254b/logstash-output-elasticsearch-10.1.0-java/lib/logstash/outputs/elasticsearch/http_client.rb:66:in `initialize'", "/opt/home//option/all/vendor/gems/71a5254b/logstash-output-elasticsearch-10.1.0-java/lib/logstash/outputs/elasticsearch/http_client_builder.rb:106:in `create_http_client'", "/opt/home//option/all/vendor/gems/71a5254b/logstash-output-elasticsearch-10.1.0-java/lib/logstash/outputs/elasticsearch/http_client_builder.rb:102:in `build'", "/opt/home//option/all/vendor/gems/71a5254b/logstash-output-elasticsearch-10.1.0-java/lib/logstash/outputs/elasticsearch.rb:238:in `build_client'", "/opt/home//option/all/vendor/gems/71a5254b/logstash-output-elasticsearch-10.1.0-java/lib/logstash/outputs/elasticsearch/common.rb:25:in `register'", "org/logstash/cooptionig/ir/compiler/OutputStrategyExt.java:106:in `register'", "org/logstash/cooptionig/ir/compiler/AbstractOutputDelegatorExt.java:48:in `register'", "/opt/home//option/all/logstash-core/lib/logstash/java_pipeline.rb:205:in `block in register_plugins'", "org/jruby/RubyArray.java:1800:in `each'", "/opt/home//option/all/logstash-core/lib/logstash/java_pipeline.rb:204:in `register_plugins'", "/opt/home//option/all/logstash-core/lib/logstash/java_pipeline.rb:495:in `maybe_setup_out_plugins'", "/opt/home//option/all/logstash-core/lib/logstash/java_pipeline.rb:217:in `start_workers'", "/opt/home//option/all/logstash-core/lib/logstash/java_pipeline.rb:159:in `run'", "/opt/home//option/all/logstash-core/lib/logstash/java_pipeline.rb:118:in `block in start'"], :thread=>"#<Thread:0x3c95bbe8@/opt/home//option/all/logstash-core/lib/logstash/java_pipeline.rb:114 run>"}
[2020-09-25T11:57:32,254][ERROR][logstash.agent           ] Failed to execute action {:id=>:main, :action_type=>LogStash::ConvergeResult::FailedAction, :message=>"Could not execute action: PipelineAction::Create<main>, action_result: false", :backtrace=>nil}

**Comment by shubhamjaincav** [comment]
any update

**Comment by shreyas29** [comment]
I am getting the same error :(

**Comment by kares** [comment]
Thanks for the report, this is definitely a bug - which goes all the way down to the [Manticore library](https://github.com/cheald/manticore/issues/87#issuecomment-779173845) and likely JRuby itself.

There's no good work-around atm except patching the manticore library (version used depends on LS version) like:
```patch
diff --git vendor/bundle/jruby/2.5.0/gems/manticore-0.7.0-java/lib/manticore/client.rb vendor/bundle/jruby/2.5.0/gems/manticore-0.7.0-java/lib/manticore/client.rb
index 2d1242d0e..604c65592 100644
--- vendor/bundle/jruby/2.5.0/gems/manticore-0.7.0-java/lib/manticore/client.rb
+++ vendor/bundle/jruby/2.5.0/gems/manticore-0.7.0-java/lib/manticore/client.rb
@@ -80,7 +80,6 @@ module Manticore
     include_package "org.apache.http.auth"
     include_package "java.util.concurrent"
     include_package "org.apache.http.client.protocol"
-    include_package "org.apache.http.conn.ssl"
     include_package "java.security.cert"
     include_package "java.security.spec"
     include_package "java.security"
@@ -90,6 +89,7 @@ module Manticore
     java_import "org.manticore.HttpGetWithEntity"
     java_import "org.manticore.HttpDeleteWithEntity"
     java_import "org.apache.http.auth.UsernamePasswordCredentials"
+    java_import "org.apache.http.conn.ssl.SSLConnectionSocketFactory"
 
     # This is a class rather than a proc because the proc holds a closure around
     # the instance of the Client that creates it.

```

Otherwise if anyone can run with the following *config/jvm.options* added to debug the underlying issue, pls give us LS std output:
```
-Djruby.log.exceptions=true
```

**Comment by kares** [comment]
Hopefully, this is resolved by now with Manticore 0.8.0 upstream release.
